.github/workflows/build_test.yaml

```diff
@@ -0,0 +1,215 @@
+      - build_base
+
+    # run this job even if build_base did not run
+    if: ${{ ! cancelled() && (needs.build_base.result == 'success' || needs.build_base.result == 'skipped') }}
+
+    strategy:
+      matrix:
+        backend:
+          - name: openmp
+            c_compiler: gcc
+            cxx_compiler: g++
+            cmake_flags: -DKokkos_ENABLE_OPENMP=ON
+          - name: cuda
+            c_compiler: gcc
+            cxx_compiler: g++
+            cmake_flags: -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_AMPERE80=ON
+          - name: hip
+            c_compiler: hipcc
+            cxx_compiler: hipcc
+            cmake_flags: -DKokkos_ENABLE_HIP=ON -DKokkos_ARCH_VEGA90A=ON
+          - name: sycl
+            c_compiler: icx
+            cxx_compiler: icpx
+            cmake_flags: -DKokkos_ENABLE_SYCL=ON -DKokkos_ARCH_INTEL_GEN=ON
+        target:
+          - name: native
+            cmake_flags: ""
+          - name: host_device
+            cmake_flags: -DKokkosFFT_ENABLE_HOST_AND_DEVICE=ON
+        exclude:
+          - backend:
+              name: openmp
+            target:
+              name: host_device
+
+    steps:
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@v1.2.0
+        with:
+          tool-cache: true
+          large-packages: false
+
+      - name: Checkout built branch
+        uses: actions/checkout@v4
+        with:
+          submodules: recursive
+
+      - name: Configure
+        run: |
+          docker run -v ${{ github.workspace }}:/work ghcr.io/cexa-project/kokkos-fft/base_${{ matrix.backend.name }}_${{ needs.check_docker_files.outputs.image_name_suffix }} \
+            cmake -B build \
+            -DCMAKE_INSTALL_PREFIX=/work/install \
+            -DCMAKE_BUILD_TYPE=${{ env.BUILD_TYPE }} \
+            -DCMAKE_C_COMPILER=${{ matrix.backend.c_compiler }} \
+            -DCMAKE_CXX_COMPILER=${{ matrix.backend.cxx_compiler }} \
+            -DCMAKE_CXX_STANDARD=17 \
+            -DBUILD_TESTING=ON \
+            -DKokkosFFT_INTERNAL_Kokkos=ON \
+            ${{ matrix.backend.cmake_flags }} \
+            ${{ matrix.target.cmake_flags }}
+
+      - name: Build
+        run: |
+          docker run -v ${{ github.workspace }}:/work ghcr.io/cexa-project/kokkos-fft/base_${{ matrix.backend.name }}_${{ needs.check_docker_files.outputs.image_name_suffix }} \
+            cmake --build build -j $(( $(nproc) * 2 + 1 ))
+
+      - name: Prepare artifacts
+        # this is mandatory to preserve execution rights
+        run: tar -cvf tests_${{ matrix.backend.name }}.tar build/
+        if: ${{ matrix.target.name == 'native' }}
+
+      - name: Save artifacts
+        # use v3 as more recent versions cannot run on Ruche
+        uses: actions/upload-artifact@v3
+        with:
+          name: tests_${{ matrix.backend.name }}
+          path: tests_${{ matrix.backend.name }}.tar
+        if: ${{ matrix.target.name == 'native' }}
+
+      - name: Install
```

Comment:
I am still not sure what is done in your Install test.

For the install test, I am expecting a user to install `KokkosFFT` on top of their pre-installed `Kokkos`. 

Accordingly, test would include following steps: 
1. Installation of `Kokkos`
2. Installation of `KokkosFFT` using `Kokkos` whose location is indicated for example by `CMAKE_PREFIX_PATH`
3. Then compile a code relying on both `Kokkos` and `KokkosFFT` as libraries whose locations are given by `CMAKE_PREFIX_PATH`

Are these steps covered in your new CI or is there some difficulty (or issue) for a test like this?

Reply:
There is indeed a difference between this workflow and the one you describe, as here we:

1. Compile and install Kokkos FFT with its embedded Kokkos library; and
2. Compile a sample code relying on Kokkos and Kokkos FFT, using `CMAKE_PREFIX_PATH` to find them.

I recognize there is a regression here.

As we discussed earlier, using an existing installation of Kokkos to build Kokkos FFT is planned for the next PR, and I will tackle this problem at once, both for building Kokkos FFT and the sample code.

Reply:
Sure. 
Thank you for the explanation. 
Anyway I need to add Kokkos version checking (#32) in another PR.